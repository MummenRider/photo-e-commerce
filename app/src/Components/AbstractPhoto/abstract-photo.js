import {
  Container,
  CenterBlock,
  Title,
  Description,
  Divider,
  ImageBlockOne,
  ImageBlockTwo,
  ImageBlockThree,
  LeftBlock,
  RightBlock,
  ImageItem,
} from "Components/AbstractPhoto/abstract-photo.styles";

export default function AbstractPhoto({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

AbstractPhoto.LeftBlock = function AbstractPhotoLeftBlock({
  children,
  ...restProps
}) {
  return <LeftBlock {...restProps}>{children}</LeftBlock>;
};

AbstractPhoto.CenterBlock = function AbstractPhotoCenterBlock({
  children,
  ...restProps
}) {
  return <CenterBlock {...restProps}>{children}</CenterBlock>;
};

AbstractPhoto.RightBlock = function AbstractPhotoRightBlock({
  children,
  ...restProps
}) {
  return <RightBlock {...restProps}>{children}</RightBlock>;
};

AbstractPhoto.ImageItem = function AbstractPhotoImageItem({
  show,
  ...restProps
}) {
  return show ? <ImageItem {...restProps} data-testid="abstract-img" /> : null;
};

AbstractPhoto.Title = function AbstractPhotoTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

AbstractPhoto.Description = function AbstractPhotoDescription({
  children,
  ...restProps
}) {
  return <Description {...restProps}>{children}</Description>;
};

AbstractPhoto.Divider = function AbstractPhotoDivider({ show, ...restProps }) {
  return show ? <Divider {...restProps} /> : null;
};

AbstractPhoto.ImageBlockOne = function AbstractPhotoImageBlockOne({
  children,
  ...restProps
}) {
  return <ImageBlockOne {...restProps}>{children}</ImageBlockOne>;
};

AbstractPhoto.ImageBlockTwo = function AbstractPhotoImageBlockTwo({
  children,
  ...restProps
}) {
  return <ImageBlockTwo {...restProps}>{children}</ImageBlockTwo>;
};

AbstractPhoto.ImageBlockThree = function AbstractPhotoImageBlockThree({
  children,
  ...restProps
}) {
  return <ImageBlockThree {...restProps}>{children}</ImageBlockThree>;
};
