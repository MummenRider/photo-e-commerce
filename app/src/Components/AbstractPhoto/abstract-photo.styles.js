import styled from "styled-components";

export const Container = styled.section`
  margin-top: 50%;
  background-color: #dbd8c7;
  display: flex;
  flex-direction: column;

  @media screen and (min-width: 768px) {
    margin-top: 20%;
    flex-direction: row;
    padding: 6% 0 25% 0;
  }

  @media screen and (min-width: 1400px) {
    padding: 6% 0 15% 0;
  }
`;

export const LeftBlock = styled.div``;

export const CenterBlock = styled.div`
  display: flex;
  flex-direction: column;

  @media screen and (min-width: 768px) {
    width: 31vw;
    padding-top: 10%;
  }
`;

export const RightBlock = styled.div`
  position: relative;
  width: 100%;

  @media screen and (min-width: 768px) {
    width: 50%;
  }
`;

export const ImageItem = styled.div`
  background-image: url(${(props) => props.src});
  background-size: cover;
  background-position: ${(props) => props.position};
  background-repeat: no-repeat;
  width: 100%;
  padding-top: ${(props) => props.padTop};
`;

export const Title = styled.h1`
  font-family: var(--ff-secondary);
  font-size: 4rem;
  font-weight: 200;
  text-align: center;

  @media screen and (min-width: 768px) {
    font-size: clamp(3rem, 2.5vw, 16rem);
  }
`;

export const Description = styled.p`
  font-family: var(--ff-primary);
  font-size: 1.2rem;
  font-weight: 200;
  line-height: 1.6;
  letter-spacing: 1px;
  text-align: center;
  padding: 5% 10% 10% 10%;

  @media screen and (min-width: 768px) {
    font-size: clamp(0.8rem, 0.9vw, 6rem);
  }

  @media screen and (min-width: 1100px) {
    padding: 10%;
  }

  @media screen and (min-width: 1100px) {
    padding: 10% 15%;
  }
`;
export const Divider = styled.div`
  position: relative;
  left: 30%;

  margin: 20% 0 15% 0;
  min-height: 0.1%;
  width: 40%;
  background-color: #949494;
  transform: rotate(-45deg);
`;

export const ImageBlockOne = styled.div`
  position: absolute;
  top: -42%;
  right: 0%;
  width: 12vw;
`;

export const ImageBlockTwo = styled.div`
  position: relative;
  top: -20%;
  left: 50%;
  transform: translate(-50%, -20%);
  width: 50vw;

  @media screen and (min-width: 768px) {
    transform: none;
    top: 12%;
    left: 0;
    width: 30vw;
    padding-left: 10%;
  }
`;

export const ImageBlockThree = styled.div`
  position: absolute;
  right: 0;
  bottom: -60%;
  width: 13vw;

  @media screen and (min-width: 1400px) {
    right: -30%;
    bottom: -32%;
  }
`;
