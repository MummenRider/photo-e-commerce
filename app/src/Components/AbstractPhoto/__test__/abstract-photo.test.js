import { render } from "@testing-library/react";
import AbstractPhoto from "Components/AbstractPhoto/abstract-photo";
import p1 from "Assets/Images/abstract_img_1.jpg";
import p2 from "Assets/Images/abstract_img_2.jpg";
import p3 from "Assets/Images/abstract_img_3.jpg";
import p4 from "Assets/Images/abstract_img_4.jpg";

describe("<AbstractPhoto />", () => {
  it("should render <AbstractPhoto /> with populated data with image all visible", () => {
    const { container, getByText, queryAllByTestId } = render(
      <AbstractPhoto>
        <AbstractPhoto.LeftBlock>
          <AbstractPhoto.ImageBlockTwo>
            <AbstractPhoto.ImageItem src={p2} show={true} />
            <AbstractPhoto.ImageBlockOne>
              <AbstractPhoto.ImageItem src={p1} show={true} />
            </AbstractPhoto.ImageBlockOne>
            <AbstractPhoto.ImageBlockThree>
              <AbstractPhoto.ImageItem src={p3} show={true} />
            </AbstractPhoto.ImageBlockThree>
          </AbstractPhoto.ImageBlockTwo>
        </AbstractPhoto.LeftBlock>

        <AbstractPhoto.CenterBlock>
          <AbstractPhoto.Title>This is a title</AbstractPhoto.Title>
          <AbstractPhoto.Divider show={true} />
          <AbstractPhoto.Description>
            Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam
            nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam
            erat, sed diam voluptua. At vero eos et accusam et justo duo dolores
            et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est
            Lorem ipsum dolor sit amet.
          </AbstractPhoto.Description>
        </AbstractPhoto.CenterBlock>

        <AbstractPhoto.RightBlock>
          <AbstractPhoto.ImageItem src={p4} position={"initial"} show={true} />
        </AbstractPhoto.RightBlock>
      </AbstractPhoto>
    );

    expect(getByText("This is a title")).toBeTruthy();
    expect(getByText(/Lorem/i)).toBeTruthy();
    expect(queryAllByTestId("abstract-img")).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it("should not render image when show is set to false", () => {
    const { container, queryByTestId } = render(
      <AbstractPhoto.ImageItem show={false} />
    );

    expect(queryByTestId("abstract-img")).toBeFalsy();
    expect(container).toMatchSnapshot();
  });
});
