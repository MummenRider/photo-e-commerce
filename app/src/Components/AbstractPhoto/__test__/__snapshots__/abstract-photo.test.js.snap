// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AbstractPhoto /> should not render image when show is set to false 1`] = `<div />`;

exports[`<AbstractPhoto /> should render <AbstractPhoto /> with populated data with image all visible 1`] = `
<div>
  <section
    class="sc-bdnxRM kuFmtL"
  >
    <div
      class="sc-gtsrHT"
    >
      <div
        class="sc-pNWdM dUaizU"
      >
        <div
          class="sc-eCApnc hziMjk"
          data-testid="abstract-img"
          src="abstract_img_2.jpg"
        />
        <div
          class="sc-fujyAs girNZW"
        >
          <div
            class="sc-eCApnc idtfEv"
            data-testid="abstract-img"
            src="abstract_img_1.jpg"
          />
        </div>
        <div
          class="sc-jrsJWt jmLeCP"
        >
          <div
            class="sc-eCApnc hDhIUh"
            data-testid="abstract-img"
            src="abstract_img_3.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-dlnjwi dpOBYy"
    >
      <h1
        class="sc-jSFjdj kTgxSH"
      >
        This is a title
      </h1>
      <div
        class="sc-iCoGMd bNWLdH"
      />
      <p
        class="sc-gKAaRy dfECc"
      >
        Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet.
      </p>
    </div>
    <div
      class="sc-hKFxyN bFxYLe"
    >
      <div
        class="sc-eCApnc jneklQ"
        data-testid="abstract-img"
        src="abstract_img_4.jpg"
      />
    </div>
  </section>
</div>
`;
